src/apps/components/AppPage/AppPage.tsx

```diff
@@ -24,38 +26,48 @@ export const AppPage: React.FC<AppPageProps> = ({
   url,
   onError,
   refetch,
-}) => (
-  <DetailPageLayout gridTemplateColumns={1} withSavebar={false}>
-    <AppPageNav
-      goBackUrl={AppUrls.resolveAppListUrl()}
-      appId={data.id}
-      name={data?.name}
-      supportUrl={data?.supportUrl}
-      homepageUrl={data?.homepageUrl}
-      author={data?.author}
-      appLogoUrl={data?.brand?.logo.default}
-    />
-    <DetailPageLayout.Content>
-      <Box
-        position="relative"
-        // It removes extra space between iframe and container
-        __lineHeight={0}
-        height="100%"
-        __minHeight={`calc(100vh - ${borderHeight} - ${topBarHeight})`}
-      >
-        {url && data?.id && data?.accessToken && (
-          <AppFrame
-            src={url}
-            appToken={data?.accessToken ?? ""}
-            onError={onError}
-            appId={data?.id ?? ""}
-            refetch={refetch}
-          />
-        )}
-      </Box>
-    </DetailPageLayout.Content>
-  </DetailPageLayout>
-);
+}) => {
+  const shop = useShop();
+
+  if (!shop?.version) {
+    return null; // loading?
```

Comment:
**suggestion**: what about converting this to an issue - if it is a problem 🤔

Reply:
I will revisit this

Reply:
https://github.com/saleor/apps/issues/682